spec/integration/app_spec.rb

```diff
@@ -0,0 +1,111 @@
+    it "shows the homepage" do
+      response = get("/")
+      expect(response.body).to include('New to Chitter? <a href="/sign_up"> Sign up here</a>')
+    end
+  end
+
+  context "GET /feed" do
+    it "shows all the peeps" do
+      response = get("/feed")
+      expect(response.body).to include("Erykah Badu")
+      expect(response.body).to include("BagLady")
+      expect(response.body).to include("I pick my friends like I pick fruit")
+      expect(response.body).to include("2023-01-08 04:05:06")
+    end
+  end
+
+  describe "Creating a new user" do
+    context "GET /sign_up" do
+      it "should get the form to create a new user" do
+        response = get("/sign_up")
+        expect(response.status).to eq(200)
+        expect(response.body).to include('<input type="text" name="name" /><br />')
+        expect(response.body).to include('<input type="password" name="password" /><br />')
+      end
+    end
+
+    context "POST /users" do
+      it "creates new user" do
+        response = post('/users', name: 'Solána Imani Rowe', username: 'sza', email: 'sza@gmail.com', password: 'killedmyex')
+        expect(response.status).to eq(200)
+        expect(response.body).to include('<h1>Welcome to the chitter community!</h1>')
+      end
+    end
+  end
+
+  describe "Logging in to an account" do
+    context "GET /login" do
+      it "should get the form to login to account" do
+        response = get("/login")
+        expect(response.status).to eq(200)
+        expect(response.body).to include('<input type="text" name="email" placeholder="Email" />')
+        expect(response.body).to include('<input type="password" name="password" placeholder="Password" />')
+      end
+    end
+
+    context "POST /login" do 
+      it "logs in a user" do 
+        _response = post('/login', email: 'erykah@gmail.com', password: 'Baduizm99')
+        response = get('/login_home')
+        expect(response.status).to eq(200)
+        expect(response.body).to include('<h3>Start peeping now</h3>')
+      end
+
+      it "fails to log in a user if wrong email or password are given" do
+        response = post('/login', email: 'erykah@gmail.co', password: 'Baduizm')
+        expect(response.status).to eq(404)
+        expect(response.body).to include('<h1>Login failed, please try again</h1>')
+      end
+    end
+  end
+
+
+  describe "Creating a new peep" do
+    context "GET /login_home" do
+      it "shows the form to add a new peep" do
+        response = get('/login_home')
+        expect(response.status).to eq(200)
+        expect(response.body).to include('<form action="/peeps" method="post">')
+        expect(response.body).to include(
+          '<input type="submit" name="button" value="Submit" />',
+        )
+      end
+    end
+
+    context "POST /" do
+      it "should add the new peep" do
+        _response = post('/peeps', message: 'hello')
+        response = get('/login_home')
+        expect(response.status).to eq(200)
```

Comment:
This assertion doesn't verify that the new peep has been created — in case something was going wrong, we could still get a 200 OK response from `/login_home` but maybe the entry wouldn't be there (which would be a false positive) — how could you also verify that the new entry has been created in this test?